README.md

```diff
@@ -1,32 +1,279 @@
+3. Manually tweak the generated client config
+4. Run code generation (once per API x language combination of interest)
+
+Each of these steps are described in more detail below. Note: the instructions assume you are running on a
+Unix-y system; if you are using Windows, you will need to tweak the steps for yourself.
+
+### Set up prerequisites for building from source and running directly
+
+Firstly, make sure you have the
+[Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) installed.
+
+Next, you need to install `protoc` if you don't have it yet: see
+[protocol-compiler-installation](https://github.com/google/protobuf#protocol-compiler-installation).
+The pre-built binaries are the easiest. There aren't instructions for installation;
+you just have to download the correct binaries for your platform and put them on your path.
+Given a platform ${PROTOC_PLATFORM} and a protoc version ${PROTOC_VERSION}:
+
+1. Download the latest zip from https://github.com/google/protobuf/releases ending in "-${PROTOC_PLATFORM}.zip"
+2. Unzip the file: `unzip protoc-${PROTOC_VERSION}-${PROTOC_PLATFORM}.zip -d protoc-${PROTOC_VERSION}`
+3. Copy `protoc` to a location on your `$PATH`: `cp protoc-${PROTOC_VERSION}/bin/protoc /usr/local/bin`
+
+Clone the present repository and build API Client Generator:
+
+```
+git clone https://github.com/googleapis/toolkit.git
+cd toolkit
+./gradlew fatJar
+```
+
+The `googleapis/toolkit` directory will hereafter be referenced as `${TOOLKIT_DIR}`.
+
+Clone the googleapis/googleapis repository (it has some config that is needed):
+
+```
+git clone https://github.com/googleapis/googleapis
+```
+
+The `googleapis/googleapis` directory will hereafter be referenced as `${GOOGLEAPIS_DIR}`.
+
+### Generate a descriptor file from the proto
+
+You need to locate/decide on the following before you can generate the descriptor file:
+
+1. The include directory with the proto files bundled with `protoc` (from the `protoc` setup step).
+   Hereafter, this will be referenced as `${PROTOC_INCLUDE_DIR}`.
+2. Any directories containing protos that your API depends on. For Google APIs defined in
+   [googleapis](https://github.com/googleapis/googleapis), this will be `${GOOGLEAPIS_DIR}`.
+3. The directory containing your proto. Hereafter, this will be referenced as `${YOUR_PROTO_DIR}`.
+4. Your proto file (or files). Hereafter, this will be referenced as `${YOUR_PROTO_FILE}`.
+4. The output file name to contain the descriptor file. Hereafter, this will be referenced as
+   `${YOUR_DESCRIPTOR_FILE}`.
+
+Run the following command to generate the proto descriptor file:
+
+```
+protoc -I=${PROTOC_INCLUDE_DIR} -I=${GOOGLEAPIS_DIR} -I=${YOUR_PROTO_DIR} \
+  --include_imports --include_source_info -o ${YOUR_DESCRIPTOR_FILE} ${YOUR_PROTO_FILE}
+```
+
+### Generate proto message classes
+
+*(Skip this section for Node.js - it loads proto files into memory at runtime.)*
```

Comment:
s/ - /— /

Reply:
But... will anyone care or notice besides you?